server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/CompareAction.java

```diff
@@ -218,4 +219,13 @@ private void writeDifferences(JsonWriter json, Map<RuleKey, ActiveRuleDiff> modi
     json.endArray();
   }
 
+  private enum RuleDtoToRuleKey implements Function<RuleDto, RuleKey> {
+    INSTANCE;
+
+    @Override
+    public RuleKey apply(@Nonnull RuleDto input) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The "input" parameter nullability is different in the superclass method, and that should not be changed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2638)